src/content/docs/new-relic-solutions/best-practices-guides/full-stack-observability/synthetic-monitoring-best-practices-guide.mdx

```diff
@@ -267,6 +267,96 @@ console.log("Duration: " + script_duration)
+    6. Set a notification method to alert your team when performance incidents occur.
+    7. You are now ready to write your script. See this example script to test making an API request and processing the response.
+
+```js
+/**
+ * Script Name: Best Practices - Node 16.10.0
+ * Author:      New Relic
+ * Version:     1.0
+ * Purpose:     Best practices example of a New Relic Synthetics scripted API monitor with async/await, secure credentials for authentication,
+ *              assert on statusCode, retries, error handling, json parsing, $util.insights, and a custom header.
+ * Reference:   https://docs.newrelic.com/docs/synthetics/synthetic-monitoring/scripting-monitors/write-synthetic-api-tests/
+ */
+
+const MAX_RETRIES = 3
+const RETRY_DELAY_MS = 200 // Delay of 200 milliseconds between retries
+
+var assert = require("assert")
+
+const makeGetRequest = async (url, options, maxRetries) => {
+  for (let attempt = 1; attempt <= maxRetries; attempt++) {
+    try {
+      console.log(`Attempt ${attempt} of ${maxRetries}`)
+      const response = await $http.get(url, options)
+
+      // Use assert to check for status code 200.
+      // This will throw an error if the status code is not 200.
+      assert.equal(response.statusCode, 200, "Expected HTTP status code 200")
+      console.log("Request succeeded:", response.statusCode)
+
+      // Parse the response body as JSON:
+      const jsonData =
+        typeof response.body === "string"
+          ? JSON.parse(response.body)
+          : response.body
+
+      console.log("Parsed JSON data:", jsonData)
+      return jsonData // Return parsed JSON data for further processing, if needed
+    } catch (error) {
+      console.error(`Attempt ${attempt} failed:`, error.message || error)
+      if (attempt === maxRetries) {
+        // If this was the last attempt, rethrow to be caught in the outer try/catch block
+        throw error
+      }
+      await new Promise((resolve) => setTimeout(resolve, RETRY_DELAY_MS))
+    }
+  }
+}
+
+;(async () => {
+  try {
+    // Get a list of applications in New Relic
+    const url = "https://api.newrelic.com/v2/applications.json"
+
+    // Retrieve the stored secure credential
+    const apiKey = $secure.API_KEY
+
+    // Define custom headers, including the secure credential for the API key
+    const headers = {
+      "X-Api-Key": apiKey,
+      "Custom-Header": "CustomValue", // Example of a custom header
+    }
+
+    // Convey the headers as part of the $http.get request options
+    const options = {
+      headers: headers,
+    }
+
+    await makeGetRequest(url, options, MAX_RETRIES)
+
+    // This shows up in SyntheticCheck as `custom.Success`
+    $util.insights.set("Success", true)
+  } catch (error) {
+    console.error("No more retries or assertion failed:", error)
+    // This shows up in SyntheticCheck as `custom.Error` if an error is caught
+    $util.insights.set("Error", error.message || "Error without a message")
```

Comment:
The error message would be included by default on the SyntheticCheck event. Why do we need to add a custom attribute for it?

Reply:
Just an example of how to use `$util.insights.set()`.